### PR TITLE
Avoid "excessive message size" for session tickets

### PR DESCRIPTION
**Pull request by mattcaswell**
We received a report of an "excessive message size" for a received session ticket. Our maximum size was significantly less than the theoretical maximum. The server may put any data it likes in the session ticket including (for example) the full certificate chain so we should be able to handle longer tickets. Update the value to the maximum allowed by the spec.

I consider this a bug which should be applied to both master and 1.1.1 (it cherry-picks cleanly).

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Good for both 1.1.1 and 3.0.

File: ssl/statem/statem_clnt.c [hunk 1]
```diff
@@ -996,7 +996,10 @@ size_t ossl_statem_client_max_message_size(SSL *s)
         return CCS_MAX_LENGTH;
 
     case TLS_ST_CR_SESSION_TICKET:
-        return SSL3_RT_MAX_PLAIN_LENGTH;
+        if (SSL_IS_TLS13(s))
+            return SESSION_TICKET_MAX_LENGTH_TLS13;
+        else
```
Comment (richsalz) [inline ssl/statem/statem_clnt.c:1001]: really hate the else after return.  maybe use `?:` ?

**Comment by mattcaswell** [comment]
Fixup pushed to avoid return after else. @paulidale please reconfirm.

**Comment by paulidale** [comment]
Still good.

**Comment by paulidale** [comment]
Merged.

**Comment by mattcaswell** [comment]
This was approved for master and 1.1.1 but only merged to master. Now, also pushed to 1.1.1 in 5b71c6777.

**Pull request closed without merging**